### PR TITLE
Started writing script logic.

### PR DESCRIPTION
**Pull request by fredericpoitevin**

### CODE REVIEW

File: scripts/main.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    if not conditional_mkdir(config.root_dir):
+        print(f"Error: cannot create root path.")
+        return -1
+
+    shutil.copy2(config_filepath, config.root_dir)
+
+    if(config.task == 'test'):
+        test(config)
+    elif(config.task == 'make_powder'):
+        make_powder(config)
```
Comment (apeck12) [inline scripts/main.py:28]: Is something along the lines of `eval(config.task(config))` what you were thinking of to simplify the logic here? I think that should enable parsing a string as a function.

Comment (fredericpoitevin) [reply scripts/main.py:28]: Yes! Sounds exactly like it!

Comment (apeck12) [reply scripts/main.py:29]: I played around with the `eval` function in Jupyter, but for some reason I couldn't get it to work in `main.py`. So for now I've kept this as is, but will try to revisit it later.

**Comment by apeck12** [comment]
`make_powder` successfully deployed on psanaq using the following: `. elog_submit.sh -c ../examples/make_powder.yaml`.

A few notes:
1. I had to make `main.py` executable; if those permissions settings aren't tracked by Github, perhaps we should add `chmod +x main.py` to elog_submit.sh?
2. I've set the default number of cores to 1 for now, but making a note here to update it to use MPI by default once parallelization is in.
3. Let's chat about the directory structure of the output? Would it be better to have a series of runX/max_powder.npy or powders/max_powder_runX.npy? The latter seems more efficient and is consistent with the current `make_powder.yaml`, in which case I'll go back and update the nomenclature for saving the powders.
4. The `slurm*out` file contains a lot of details, e.g.:
```
...
DEBUG:Detector.UtilsEpix10ka:common-mode correction pars cmp: [ 7.  2. 10. 10.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
DEBUG:Detector.UtilsEpix10ka:gmask shape:(16, 352, 384) size:2162688 dtype:int8 [1 1 1 1 1...]
  per panel statistics of cm-corrected pixels: [135168 135151 135167 135168 135165 135168 135168 135168 135167 135164
 135168 135168 135160 135167 135168 135168]
DEBUG:Detector.UtilsEpix10ka:TIME common-mode correction = 0.183689 sec for cmp=[ 7.  2. 10. 10.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.]
...
``` 
which may be hard to sift through in case of an error.

**Pull request merged**